ppa/archive/management/commands/update_hathi_pagecounts.py

```diff
@@ -0,0 +1,87 @@
+    By default, runs on all non-excerpted, public HathiTrust items.
+    """
+
+    help = __doc__
+
+    #: normal verbosity level
+    v_normal = 1
+    verbosity = v_normal
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "source_ids", nargs="*", help="List of specific items to update (optional)"
+        )
+
+    def handle(self, *args, **kwargs):
+        verbosity = kwargs.get("verbosity", self.v_normal)
+        source_ids = kwargs.get("source_ids", [])
+        # page count does not affect solr indexing, so disconnect signal handler
+        IndexableSignalHandler.disconnect()
+
+        script_user = User.objects.get(username=settings.SCRIPT_USERNAME)
+        digwork_contentype = ContentType.objects.get_for_model(DigitizedWork)
+
+        # find all non-excerpted, non-suppressed hathi volumes
+        hathi_vols = DigitizedWork.objects.filter(
+            source=DigitizedWork.HATHI,
+            item_type=DigitizedWork.FULL,
+            status=DigitizedWork.PUBLIC,
+        )
+        # if source ids are specified, limit to those records only
+        if source_ids:
+            hathi_vols = hathi_vols.filter(source_id__in=source_ids)
+
+        stats = defaultdict(int)
+
+        for digwork in hathi_vols:
+            try:
+                # store the current page count
+                old_page_count = digwork.page_count
+                # recalculate page count from pairtree data
+                # NOTE: this method automatically saves if page count changes
+                digwork.page_count = digwork.count_pages()
+                if old_page_count != digwork.page_count:
+                    stats["updated"] += 1
+                    # create a log entry documenting page count change
+                    LogEntry.objects.log_action(
+                        user_id=script_user.pk,
+                        content_type_id=digwork_contentype.pk,
+                        object_id=digwork.pk,
+                        object_repr=str(digwork),
+                        change_message=f"Recalculated page count (was {old_page_count}, "
+                        + f"now {digwork.page_count})",
+                        action_flag=CHANGE,
+                    )
+
+                else:
+                    stats["unchanged"] += 1
+
+            except storage_exceptions.ObjectNotFoundException:
+                if verbosity >= self.v_normal:
+                    self.stderr.write(
+                        self.style.WARNING(f"Pairtree data for {digwork} not found")
+                    )
+                stats["missing_data"] += 1
+
+        # report a summary of what was done
+        if verbosity >= self.v_normal:
+            self.stdout.write(
+                f"""Volumes with updated page count: {stats['updated']:,}
+    Page count unchanged: {stats['unchanged']:,}
+    Missing pairtree data: {stats['missing_data']:,}"""
+            )
```

Comment:
I'm unfamiliar with how spacing for logging is handled, but might it be better to use explicit tabs `\t` instead of spaces? This would improve the file formatting and remove the dependency on triple quotes.